### PR TITLE
Add JSDocs for IncludeCallback

### LINKED ISSUE #144: Missing JSDocs for IncludeCallback
`ejs.compile(str, {client: true}` returns a [`ClientFunction`](http://ryanzim.github.io/ejs-api/global.html#ClientFunction) with the usage: `ClientFunction(locals, escape, include, rethrow)`, according to the jsdocs. The [`EscapeCallback`](http://ryanzim.github.io/ejs-api/global.html#EscapeCallback) and [`RethrowCallback`](http://ryanzim.github.io/ejs-api/global.html#RethrowCallback) are documented, the `IncludeCallback` is not.

I personally haven't been able to get it working; could someone explain and/or document it's parameters?

### PR DESCRIPTION
**Pull request by RyanZim**
Fixes #144 

Sorry for the messy diff, I re-ordered the callbacks in template-functions.jsdoc and git didn't "get it."

Any objections or suggestions @mde or @TimothyGu?

**If nothing is said within two days, I will merge this.**

### CODE REVIEW

File: docs/jsdoc/template-functions.jsdoc [hunk 1]
```diff
@@ -49,11 +59,12 @@
...
 /**
- * Escapes a string using HTML/XML escaping rules.
+ * The callback called by ClientFunction to include files at runtime with `include()`
  *
- * @callback EscapeCallback
- * @param  {String} markup Input string
- * @return {String}        Escaped string
+ * @callback IncludeCallback
+ * @param  {String} path      path to be included
+ * @param  {Object} [data]    data passed to the template
```
Comment (TimothyGu) [inline docs/jsdoc/template-functions.jsdoc]: Inconsistent spelling

File: docs/jsdoc/template-functions.jsdoc [hunk 2]
```diff
@@ -49,11 +59,12 @@
  */
 
 /**
- * Escapes a string using HTML/XML escaping rules.
+ * The callback called by ClientFunction to include files at runtime with `include()`
```
Comment (TimothyGu) [inline docs/jsdoc/template-functions.jsdoc]: Is `ClientFunction` automatically linked?

**Comment by TimothyGu** [comment]
Sorry for being late. Feel free to fix these nits by directly pushing.

File: docs/jsdoc/template-functions.jsdoc (hunk 1 above)
Comment (RyanZim) [inline docs/jsdoc/template-functions.jsdoc]: What's inconsistent? :confused:  I can't see anything out of line.

Probably I'm just missing something.

File: docs/jsdoc/template-functions.jsdoc (hunk 2 above)
Comment (RyanZim) [inline docs/jsdoc/template-functions.jsdoc]: No. I meant to add a `@link` but somehow I must have forgotten.

File: docs/jsdoc/template-functions.jsdoc (hunk 1 above)
Comment (TimothyGu) [inline docs/jsdoc/template-functions.jsdoc]: Sorry I meant capitalization.

```diff
- * @param  {String} path      path to be included
+ * @param  {String} path      Path to be included
```

Comment (RyanZim) [inline docs/jsdoc/template-functions.jsdoc]: OK, thanks

**Comment by RyanZim** [comment]
Fixed in a78de100535bdf15d246e04c87da3a7cb80f271d.

**Pull request merged**